### DISCUSSION TITLE
Troubleshoot sessions

### DISCUSSION THREAD

**Discussion by Strykar** [kovidgoyal/kitty#4066] [Q&A] [answered]
I start kitty with a session file that opens multiple tabs, names them and runs the programs for each tab and this works well.
However, every now and then Kitty will not open one of the tabs I have listed in its invoked session file and while the exact same commands for that tab run fine in a new tab inside Kitty, I have no way of troubleshooting why the tab was not opened and the commands not run.

The tabs before and after the offending tab open fine, is there a way to test a session file or have kitty log why it ignored / did not start a particular tab?

**Comment by kovidgoyal** [comment, marked as answer]
All errors are logged to stdout/stderr. Just run kitty from a script
that redirects them to some log file.

**Comment by Strykar** [comment]
Started kitty as: `/usr/bin/kitty --instance-group=1 --config=/home/strykar/.config/kitty/weechatkitty.conf --session=/home/strykar/.config/kitty/weechat.session --listen-on=unix:/tmp/mykitty --directory=/home/strykar &> /home/strykar/.config/kitty/kitty.log`

This is all that is logged: 
`tail -f ~/.config/kitty/kitty.log
[270 16:47:31.139872] Failed to send resize signal to child with id: 6 (children count: 6) (add queue: 0)`

**Comment by kovidgoyal** [comment]
You need to redirect stderr as well

On Mon, Sep 27, 2021 at 04:21:59AM -0700, Avinash H. Duduskar wrote:
> Started kitty as: `/usr/bin/kitty --instance-group=1 --config=/home/strykar/.config/kitty/weechatkitty.conf --session=/home/strykar/.config/kitty/weechat.session --listen-on=unix:/tmp/mykitty --directory=/home/strykar &> /home/strykar/.config/kitty/kitty.log`
> 
> This is all that is logged: 
> `tail -f ~/.config/kitty/kitty.log
> [270 16:47:31.139872] Failed to send resize signal to child with id: 6 (children count: 6) (add queue: 0)`
> 
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/4066#discussioncomment-1389769

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by Strykar** [comment]
Run as `/usr/bin/bash -c '/usr/bin/kitty --instance-group=1 --config=/home/strykar/.config/kitty/weechatkitty.conf --session=/home/strykar/.config/kitty/weechat.session --listen-on=unix:/tmp/mykitty --directory=/home/strykar >> /home/strykar/.config/kitty/kitty.log 2>&1'`

```
$ tail -f /home/strykar/.config/kitty/kitty.log
[270 17:00:46.652704] [PARSE ERROR] Unknown CSI code: 'm' with start_modifier: '' and end_modifier: '%' and parameters: '0'
[270 17:08:08.456504] [PARSE ERROR] Unrecognized DCS code: 0x7a
[270 17:08:08.456558] [PARSE ERROR] Unknown CSI code: 'm' with start_modifier: '' and end_modifier: '%' and parameters: '0'
[270 19:51:08.291290] [PARSE ERROR] Unrecognized DCS code: 0x7a
[270 19:51:08.291556] [PARSE ERROR] Unknown CSI code: 'm' with start_modifier: '' and end_modifier: '%' and parameters: '0'
[270 19:51:28.478298] [PARSE ERROR] Unrecognized DCS code: 0x7a
[270 19:51:28.478343] [PARSE ERROR] Unknown CSI code: 'm' with start_modifier: '' and end_modifier: '%' and parameters: '0'
[270 19:54:58.312273] [PARSE ERROR] Unrecognized DCS code: 0x7a
[270 19:54:58.312312] [PARSE ERROR] Unknown CSI code: 'm' with start_modifier: '' and end_modifier: '%' and parameters: '0'
X connection to :1 broken (explicit kill or server shutdown).
```

**Reply by kovidgoyal** [reply]
There are no errors, which means the tab was started fine, most likely the program running in it died and so the tab was closed. If you want to debug further you will need to run kitty from source.

**Reply by Strykar** [reply]
There are two programs being run (layout vertical) and I am 100% sure they run fine because if I open another tab, they run fine.
By source you mean, compile kitty with symbols and debug via GDB backtraces?

**Reply by Strykar** [reply]
> There are no errors, which means the tab was started fine, most likely the program running in it died and so the tab was closed. If you want to debug further you will need to run kitty from source.

The program monitors another program in a previous tab, your comment got me thinking.. so I added a `sleep 20` before and lo behold, it starts the tab fine. It was indeed dying/exiting because the program it had to monitor hadn't started yet.

Is `sleep` the right way to cascade this?

**Comment by kovidgoyal** [comment]
On Mon, Sep 27, 2021 at 07:48:49AM -0700, Avinash H. Duduskar wrote:
> By source you mean, compile it with symbols and debug via GDB backtraces?

No, the kitty UI is written in python. You will need to simply run it
from source and make some modifications to confirm that the tab is being
closed because of child death. Essentially, add some more logging.

https://sw.kovidgoyal.net/kitty/build/

**Comment by kovidgoyal** [comment]
On Mon, Sep 27, 2021 at 07:58:28AM -0700, Avinash H. Duduskar wrote:
> > There are no errors, which means the tab was started fine, most likely the program running in it died and so the tab was closed. If you want to debug further you will need to run kitty from source.
> 
> The program monitors another program in a previous tab, your comment got me thinking.. so I added a sleep 20 before and lo behold, it starts the tab fine. It was indeed dying/exiting because the program it had to monitor hadn't started. Is sleep the right way to cascade this?

That's not really a kitty question, the correct way to do this would be
to design your monitoring program to wait till the program it is
monitoring has started, with some kind of max wait.

**Reply by Strykar** [reply]
Thank you! Sorry for taking up your time with this, I should have done some more troubleshooting before opening this. :pray:
